File: ARRAY/Array/Sorting_0_1/sort.js

let arr = [0, 1, 1, 0, 1, 0, 0, 1];
arr.sort((a, b) => a - b );
console.log(arr);
function sort01(arr) {
    let left = 0;
    let right = arr.length - 1

    while(left < right){

	while(left < right && arr[left] == 0){left++}
	while(left < right && arr[right] == 1){right--}
	
	if(left < right){
		arr[left] = 0;
		arr[right] = 1;
		left++
		right--
	}


}

}

File: ARRAY/Array/Sorting_0_1/sort.py

def sort01(arr):
    """
    Sorts a binary array (containing only 0s and 1s) in-place.
    
    Parameters:
    arr (list): A list of integers containing only 0s and 1s.
    
    Returns:
    None: The input list is sorted in-place.
    """
    left, right = 0, len(arr) - 1
    
    while left < right:
        if arr[left] == 0:
            left += 1
        elif arr[right] == 1:
            right -= 1
        else:
            arr[left], arr[right] = arr[right], arr[left]
            left += 1
            right -= 1
            
            
    
    